### PR TITLE
Remove privilege=true

### PR DESCRIPTION
**Pull request by andraxylia**
Added a helm option to control if privileged=true is set in istio-proxy container and istio-init container.

Setting debug to true also enables privileged mode.

For the enable-core-dump-container, privileged mode is always set to true.

### CODE REVIEW

**Comment by andraxylia** [comment]
@costinm ptal, this is ready to be merged.

File: install/kubernetes/helm/istio-remote/templates/sidecar-injector-configmap.yaml [hunk 1]
```diff
@@ -130,7 +132,6 @@ data:
           value: {{ "[[ or (index .ObjectMeta.Annotations \"sidecar.istio.io/interceptionMode\") .ProxyConfig.InterceptionMode.String ]]" }}
         imagePullPolicy: {{ .Values.global.imagePullPolicy }}
         securityContext:
-          privileged: false
```
Comment (costinm) [inline install/kubernetes/helm/istio-remote/templates/sidecar-injector-configmap.yaml]: Why did you remove this ?

**Comment by costinm** [comment]
Can you take a look at PR 7488 ? It seems NET_ADMIN_RAW is needed - but I'm not sure I understand the full explanation. If we remove priv, we may need to add NET_ADMIN_RAW ? 

Also try to keep istio-remote in sync, not a regression but we'll need to fix it. 

+1 for the rest.

File: install/kubernetes/helm/istio-remote/templates/sidecar-injector-configmap.yaml (hunk 1 above)
Comment (andraxylia) [reply install/kubernetes/helm/istio-remote/templates/sidecar-injector-configmap.yaml]: By default it is false, we do not need to set it.

Comment (andraxylia) [reply install/kubernetes/helm/istio-remote/templates/sidecar-injector-configmap.yaml]: There are other things that can be changed in the securityContext, and we do not add in the yaml the default value. This should be a best practice to keep the yaml files small.

**Comment by andraxylia** [comment]
I am looking at #7488 , we can add some of the changes there on top of this, so let's get this in first.

**Comment by andraxylia** [comment]
I made the same type of changes in istio-remote.

**Comment by andraxylia** [comment]
There is no sufficient explanation or proof NET_ADMIN_RAW is needed, and I will not add it.

We had Istio working before without NET_ADMIN_RAW @ayj .

**Comment by andraxylia** [comment]
cc @louiscryan

**Comment by costinm** [comment]
The problem is that according to the test in 7448, without NET_RAW on some linux versions we won't capture ( permission denied ). Not clear which linux versions - but it would be a regression. 
May be safer to wait for 1.0.2 so we can do more testing.

File: install/consul/templates/istio.yaml.tmpl [hunk 2]
```diff
@@ -23,7 +23,6 @@ services:
           - apiserver
     ports:
       - "8080:8080"
-    privileged: true
```
Comment (rshriram) [inline install/consul/templates/istio.yaml.tmpl]: Don’t remove this. This is for docker compose.

**Comment by andraxylia** [comment]
See here for NET_RAW abuse, the Docker team recommends dropping NET_RAW:

https://www.nccgroup.trust/globalassets/our-research/us/whitepapers/2016/june/abusing-privileged-and-unprivileged-linux-containers.pdf

**Comment by andraxylia** [comment]
If privilege is really needed by some Linux versions, user can enable it via the helm option this PR is adding.

If some specific platform require priviledge=true, they can add it.

**Comment by andraxylia** [comment]
See also here on why privileged has to DISABLED:
https://www.slideshare.net/antitree/docker-security-63783756

**Comment by andraxylia** [comment]
The NET_RAW capability setting default value is dictated by the container runtime. 

type SecurityContext struct {
	// The capabilities to add/drop when running containers.
	// Defaults to the default set of capabilities granted by the container runtime.
	// +optional
	Capabilities *Capabilities ...}

By default, docker container runtime has NET_RAW enabled :
https://docs.docker.com/engine/reference/run/

If anything we need a separate item to drop NET_RAW. PR 7448 needs to add it explicitly because it drops all other capabilities.

This PR is good to be merged.

File: install/consul/templates/istio.yaml.tmpl (hunk 2 above)
Comment (andraxylia) [reply install/consul/templates/istio.yaml.tmpl]: Sure, I can put this back, but are you positive docker compose needs it? Everywhere docker advises to never run containers with priv true.

Comment (andraxylia) [reply install/consul/templates/istio.yaml.tmpl]: This is already the case, the init core dump container is not enabled by default. There is also #8043 that deals with core dumps.

I will revert the consul, but do you have any reference that shows this is required? It does not make sense.

**Comment by costinm** [comment]
Ok, we'll figure it out on the other PR. 

We still add both NET_RAW and NET_ADMIN if TPROXY is set, for the sidecar container, and the long
term plan is to use TPROXY or eBPF.

Worse case, if any platform breaks they can enable the privileged back.

**Comment by hklai** [comment]
@andraxylia does this need to be cherrypicked to master?

**Comment by linsun** [comment]
yes we should!

**Comment by hklai** [comment]
@andraxylia @linsun please help create a cherrypick PR in master. I tried but there are a lot of merge conflicts that I am not comfortable to resolve.

**Comment by andraxylia** [comment]
will do.
